src/observe-tag.js

```diff
@@ -56,19 +82,35 @@ module.exports = tagFunction => {
 
 			// if an element had focus, reapply it
 			if (removedElementWithFocusData.index >= 0) {
-				const children = tagResult.querySelectorAll('*')
-
-				const elementToGiveFocus = [tagResult, ...children][removedElementWithFocusData.index]
-				elementToGiveFocus.focus()
-
-				// also try to set the selection, if there is a selection for this element
-				const hasSelectionStart = removedElementWithFocusData.selectionStart !== null && removedElementWithFocusData.selectionStart !== undefined
-				if (hasSelectionStart) {
-					elementToGiveFocus.setSelectionRange(
-						removedElementWithFocusData.selectionStart,
-						removedElementWithFocusData.selectionEnd,
-						removedElementWithFocusData.selectionDirection
-					)
+				const allElements = parentAndChildrenElements(tagResult)
+
+				// we'll look through the elements (in order of nodes closest to original index) and find a tag that matches.
+				// this means if it didn't move, we'll get it right away,
+				// if it did, we'll look at the elements closest to the original position
+				const nodeMatchesTagName = node => node.tagName === removedElementWithFocusData.tagName
+				const elementToGiveFocus = allElements
+					.map(toIndicies)
+					.sort(byDistanceFromIndex(removedElementWithFocusData.index))
+					.map(toNodes(allElements))
+					.find(nodeMatchesTagName)
+
+				// if the element / child exists, focus it
+				if (elementToGiveFocus !== undefined) {
+					elementToGiveFocus.focus()
+
+					// also try to set the selection, if there is a selection for this element
+					const hasSelectionStart = removedElementWithFocusData.selectionStart !== null && removedElementWithFocusData.selectionStart !== undefined
```

Comment:
Is there a reason why you can check to see if `removedElementWithFocusData.selectionStart` is truthy? Do null and undefined come back as truthy  in this case, or is there a falsy value that you want to return as true?

Reply:
`selectionStart` can be `0` which is a falsy value, so yeah, we have to explicitly check if it's null or undefined.